### PR TITLE
changes for convergences

### PR DESCRIPTION
**Pull request by weijjia**
changes made for convergences:
1. replaces resources with scopes
2. update sample to test the workflow. 
3. update the unit tests, remove the tests for client-credential and username-password as they are not supported in convergence yet.

### CODE REVIEW

**Comment by msotclas** [comment]
Hi __@weijjia__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla2.msopentech.com.
    </span>

TTYL, MSOTBOT;

File: lib/authentication-context.js [hunk 1]
```diff
@@ -203,16 +203,16 @@ AuthenticationContext.prototype._acquireToken = function(callback, tokenFunction
 };
 
 /**
- * Gets a token for a given resource.
- * @param {string}   resource                            A URI that identifies the resource for which the token is valid.
+ * Gets a token for given scope.
+ * @param {string}   scope                               A URI that identifies the scope for which the token is valid.
```
Comment (omercs) [inline lib/authentication-context.js]: Is this supposed to be array or string? Implementation and header are not matching here.

File: lib/authentication-context.js [hunk 2]
```diff
@@ -220,74 +220,24 @@ AuthenticationContext.prototype.acquireToken = function(resource, userId, client
...
-  });
-};
-
-/**
- * Gets a token for a given resource.
+ * Gets a token for  given scope.
  * @param {string}   authorizationCode                   An authorization code returned from a client.
  * @param {string}   redirectUri                         The redirect uri that was used in the authorize call.
- * @param {string}   resource                            A URI that identifies the resource for which the token is valid.
+ * @param {string}   scope                               A URI that identifies the scope for which the token is valid.
```
Comment (omercs) [inline lib/authentication-context.js]: Please update scope definition in other methods as well.

File: lib/cache-driver.js [hunk 3]
```diff
@@ -515,4 +521,65 @@ CacheDriver.prototype.add = function(entry, callback) {
...
+           var potentialEntry = potentialEntries[0]; 
+           var newEntry = _.clone(potentialEntry);
+           newEntry = _.extend(newEntry, entry); 
+           self._replaceEntry(potentialEntry, newEntry, callback);
+        }
+     });
+};
+
+/**
+ * Checks to see if given entries contain the given list of scopes.
```
Comment (omercs) [inline lib/cache-driver.js]: "Checks to see if given entry" it is checking single entry here.

File: lib/cache-driver.js [hunk 4]
```diff
@@ -203,9 +203,14 @@ CacheDriver.prototype._loadSingleEntryFromCache = function(query, callback) {
     var returnVal;
     var isResourceSpecific;
 
+    // For given client id, user id, and authority, there should be only one returned. With Convergence, scope will be stored instead of resource. 
     if (potentialEntries && 0 < potentialEntries.length) {
-      var resourceSpecificEntries;
-      resourceSpecificEntries = _.where(potentialEntries, { resource : self._resource });
+      var resourceSpecificEntries = [];
+      for (var index = 0; index < potentialEntries.length; ++index) { 
+         if (_entryContainsGivenScopes(potentialEntries[index], self._scope)) {
```
Comment (omercs) [inline lib/cache-driver.js]: _scopes will look better as a name since you are taking multiple scopes.

File: lib/authentication-context.js (hunk 1 above)
Comment (weijjia) [inline lib/authentication-context.js]: it should be an array, i'll make the change.

File: lib/cache-driver.js (hunk 3 above)
Comment (weijjia) [inline lib/cache-driver.js]: updated.

**Comment by msotclas** [comment]
__@weijjia__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, MSOTBOT;

File: sample/website-sample.js [hunk 5]
```diff
@@ -64,20 +104,21 @@ if (parametersFile) {
 
 if (!parametersFile) {
```
Comment (omercs) [inline sample/website-sample.js]: clean up credentials

**Comment by omercs** [comment]
Adal.net has GetAuthorizationRequestUrlAsync method to provide url for login. I dont see similar method in nodejs.

**Comment by omercs** [comment]
:shipit:

File: sample/website-sample.js (hunk 5 above)
Comment (weijjia) [inline sample/website-sample.js]: cleaned up~

**Pull request merged**